transaction-status/src/lib.rs

```diff
@@ -612,6 +612,37 @@ impl From<VersionedConfirmedBlock> for ConfirmedBlock {
     }
 }
 
+impl TryFrom<ConfirmedBlock> for VersionedConfirmedBlock {
+    type Error = &'static str;
+
+    fn try_from(block: ConfirmedBlock) -> Result<Self, Self::Error> {
+        let expected_transaction_count = block.transactions.len();
+
+        let txs: Vec<_> = block
+            .transactions
+            .into_iter()
+            .filter_map(|tx| match tx {
+                TransactionWithStatusMeta::MissingMetadata(_) => None,
+                TransactionWithStatusMeta::Complete(tx) => Some(tx),
+            })
+            .collect();
+
+        if txs.len() != expected_transaction_count {
+            return Err("some transactions miss");
```

Comment:
Can we make this more explicit? Something like "One or more transactions missing metadata"?

Reply:
I implemented it like this https://github.com/solana-labs/solana/pull/28122/commits/3ec8c632319546c571fe98e0eab191b429068fc8